torch/csrc/autograd/profiler.cpp

```diff
@@ -192,4 +193,79 @@ double Event::cuda_elapsed_us(const Event & e) {
 
 CUDAStubs::~CUDAStubs() = default;
 
+
+static jit::CodeTemplate event_template(R"(
+{
+  "name": "${name}",
+  "ph": "X",
+  "ts": ${ts},
+  "dur": ${dur},
+  "tid": ${tid},
+  "pid": "CPU Functions",
+  "args": {}
+})");
+
+
+RecordProfile::RecordProfile(std::ostream& out)
+: out_(out) {
+  init();
+}
+
+RecordProfile::RecordProfile(const std::string& filename)
+: file_(new std::ofstream(filename)), out_(*file_) {
+  init();
+}
+
+void RecordProfile::init() {
+  enableProfiler(ProfilerState::CPU);
+}
+
+RecordProfile::~RecordProfile() {
+  thread_event_lists event_lists = disableProfiler();
+  std::vector<Event*> events;
```

Comment:
This is kind of unnecessary. We could simply have this double loop inside `processEvents` and avoid allocating as many pointers as we have events

Reply:
Plus you don't even reserve memory, meaning that we will keep reallocating the storage many many times